src/store/actionCreators/saveSets.ts

```diff
@@ -72,6 +81,36 @@ export const createSaveSetIfUnique = (
   }
 };
 
+export const editSaveSet = (
+  payload: EditSetParams,
+): ThunkAction<void, RootState, null, SaveSetsActionTypes> => async (dispatch) => {
+  const { saveSetInfo, onNameConflict, onSuccess, onFail } = payload;
+
+  try {
+    const tagIsUnique = (await saveSetCountForTag(saveSetInfo.name, saveSetInfo.currentUser)) === 0;
+    if (tagIsUnique) {
+      const result = await editSaveSetTag(saveSetInfo);
+
+      if (result && result > 0) {
+        dispatch(isEditingTag(saveSetInfo));
+        onSuccess();
+        return;
+      } else {
+        onFail();
+        return;
+      }
+    }
+
+    if (onNameConflict) {
+      onNameConflict();
+    }
+
+    dispatch(failureCreate(new SaveSetNameConflictError('A set with this name already exists')));
+  } catch (error) {
+    console.error(error);
```

Comment:
Just to make sure => nothing to clean here?

Reply:
no. the edit failed thats all